### ISSUE TITLE
RFC: Onboarding 2.0

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/meta#64]
# Request for comments: Onboarding 2.0, Joe

## What are we fixing?
The onboarding email flow is currently limited. Until [very recently](https://github.com/PostHog/product-internal/pull/352) it's been unclear to many people what emails are sent when. We currently send few emails to encourage exploration and use of certain features, with no personalization based on ICPs or other criteria. We can do better. 

## What's do we do right now?
I've mapped out a visual representation of the current onboarding flow. [Link to Miro](https://miro.com/app/board/uXjVOz6q8lw=/?share_link_id=188410474504). 

<img width="1587" alt="Screenshot 2022-09-27 at 15 32 18" src="https://user-images.githubusercontent.com/84011561/192555294-fc18bf05-5618-42d8-8bdf-6cc8e567b45e.png">

@kappa90 has also got a list of transactional emails [here](https://github.com/PostHog/product-internal/pull/352), which links directly to the content. 

Both of these only reflect a _partial_ view of the potential full email experience our users get however, as these emails are sent within the product to all users. Other email systems which exist:

- Array: Sent to all users, by Mailchimp, every two weeks
- Hogmail: Sent to subscribed users, by Mailchimp, every two weeks
- G2 request: Sent to all new users per month, by Mailchimp, every month
- Post-demo Outreach: Sent to users who have had a demo, via Hubspot, after a demo

## What do competitors do?

I did a bunch of research when we looked at Onboarding 1.0 / 1.5 as project earlier this year. No reason to think it has changed since. [Full detail here](https://github.com/PostHog/posthog/issues/9784#issue-1235349158). 

In summary: 
- Amplitude sends a four-email flow. Welcome email (Day 0), Personal follow-up (Day 1), Reminder (Day 4) and final reminder (Day 7). 
- LogRocket also send a four email flow. Welcome email (Day 0), Personal follow-up (Day 4), Reminder (Day 8) and final reminder (Day 8/9). 

PostHog, in comparison, has a zero-email flow with two optional emails which are triggered by a failure to ingest events. 

## Success criteria and proposal

I'm proposing three main goals for the onboarding flow:

- Personalize email content based on ICP criteria, so we do a better job of giving users relevant information
- Use emails to encourage best practices and initial behaviours for users, e.g. invite new users
- Create a flow which directs users to explore additional features beyond insights, in particular session recording

The flow I'm proposing to achieve this would look like this. [Link to Miro](https://miro.com/app/board/uXjVOz6q8lw=/?share_link_id=264549184013):

<img width="1326" alt="Screenshot 2022-09-27 at 16 18 49" src="https://user-images.githubusercontent.com/84011561/192566889-e6046d5f-ef82-4b31-a1b9-ec20a6875ad6.png">

This flow has several improvements over the existing version:
- It includes a welcome email for first and invited users, highlighting resources to help them get started
- It offers tailored ingestion advice for new users based on ICP
- It extends the ingestion assistance path to a third step
- It includes three onboarding emails to encourage best practices and exploration

For now, I'm proposing that we encourage users to explore Session Recording specifically because it is a feature of immediate use to the most users and one we plan to bill for separately in the future. Other features (e.g. Feature Flags) may not be intuitive or desireable for many users (e.g. non engineers). 

I'm proposing that we use this flow for both PostHog Cloud and Self-Hosted, potentially with some minor copy or content variations depending on the chosen product. However, depending on how session recording is delivered as a product it may also need a new flow (or consideration for how this path would be adapted if users just want SR, cc @benjackwhite)

Session recording can also act as a good point of reference as we can monitor whether adoption increases. If it does, we can consider a 2.5 which suggests other features, such as FFs, potentially based on ICP criteria. 

**Can we personalize further, based on ICP criteria such as industry?** I've elected not to consider this for now, as it's unclear whether industries will have different product needs beyond privacy (which will be solved by the Cloud/SH decision).

## Design and delivery
This is a V2 project, so design may want to get involved in improving emails. However, I think the existing email templates are perfectly serviceable so there's no urgent design need. Maybe some new Hedgehog art, though? 

In terms of how the emails will be delivered, I think it's simplest and best to continue sending these emails through the product itself in the existing method.  

## Sprints and what's next?

Looking for general feedback on the proposed flow, with a specific question about how we can pull the relevant information from Clearbit or Hubspot to reliably inform whether the recipient is an engineer or not. If we're unable to do this consistently, we can keep the existing step either as a fallback, or just as it is. 

Following feedback, I can begin creating email copy and supporting content for these emails across 1-2 sprints. Some engineering support (I believe @kappa90 is also interested 🤞) may be needed to help get them into the product.

**Comment by abheek9** [comment]
Thanks Joe for the transparency! Would love to add Laudspeaker to the list of tools to consider - we just got approved on your app page ([link](https://github.com/PostHog/posthog-laudspeaker-app), we are [open source](https://github.com/laudspeaker/laudspeaker), backed by YC and building out all the functionality you'll find with braze/ customer io etc. We are scrappy and would love the opportunity to help you build out the flows you need!

**Comment by kappa90** [comment]
> I understand the rationale for pushing session recordings, but given we do get more and more people signing up for feature flags alone, is it possible to almost have people 'choose their own adventure' - we could ask them 'what brought you to PostHog?' and then depending on what they click, they get a different onboarding flow. Appreciate this is more a V2 idea rather than MVP.

I love this. People could select what they would like to integrate, and then the onboarding flow could change based on that, e.g. once you have integrated events, they could have optional (very short) flows that explain how to setup SR or FF.

**Comment by joethreepwood** [comment]
[Took a first stab at content yesterday,](https://docs.google.com/document/d/1G_M3qYlpQqvQXMjxnXMTPfKvVS88UGhiPN0aeUyvoT4/edit?usp=sharing) so we can start mocking things up. Feedback welcome.

**Comment by joethreepwood** [comment]
After getting some feedback, making copy updates to reflect the now-removed demo and some awesome art assistance from @lottiecoxon, I've now finalized the copy which we'll use for the emails and I believe @smallbrownbike has finalized the layouts too. 

[I've dropped images into the doc for now, to illustrate what it'll look like. 
](https://docs.google.com/document/d/1G_M3qYlpQqvQXMjxnXMTPfKvVS88UGhiPN0aeUyvoT4/edit?usp=sharing)
Next step will be to build these in Customer.io.

@kappa90 @raquelmsmith If I put together a flow in Customer.io by Monday, could one of you help me deactivate the old flows and confirm the new one is configured correctly?
